src/MapUtil/MapUtil.js

```diff
@@ -330,7 +330,7 @@ export class MapUtil {
       const queryString = UrlUtil.objectToRequestString(
         Object.assign(params, extraParams));
 
-      return `${url}?${queryString}`;
+      return url.endsWith('?') ? `${url}${queryString}` : `${url}?${queryString}`;
```

Comment:
How about an URL like this:

```js
var url = 'https://example.com/my-mapserver?map=foo.map'
```

I this this isn't not handled correctly right now, right?

Reply:
You are right. PR is welcome as this is already merged :-)